Wingstop_Recovery_Computer_V1.ino

```diff
@@ -0,0 +1,525 @@
+      turn_on_led(); //Turn on LED
+      turn_on_buzzer(); //Turn on buzzer
+      delay(200);
+      turn_off_led(); //Turn off LED
+      turn_off_buzzer(); //Turn off buzzer
+      delay(100);
+    }
+  }
+
+
+  //FILE Setup***************************************************
+
+  open_file(); //File for writing
+  set_header_file(); //sets headers at beginning of file
+
+
+// Barometer setup********************************************************
+
+  while (!bmp.begin_I2C()) {   
+    //turn on buzzer if not connecting to barometer
+    turn_on_buzzer();
+  }
+  turn_off_buzzer();
+
+  // Set up oversampling and filter initialization settings
+  bmp.setTemperatureOversampling(BMP3_OVERSAMPLING_8X);
+  bmp.setPressureOversampling(BMP3_OVERSAMPLING_4X);
+  bmp.setIIRFilterCoeff(BMP3_IIR_FILTER_COEFF_1);//no filtering
+  bmp.setOutputDataRate(BMP3_ODR_50_HZ);
+
+
+
+for(int i = 1; i<10; i++){ //calibrates initial pressure and starting altitude to zero
+  read_barometer();//updates barometer data
+  delay(50);
+
+  //calibrate readings for altitude measurements. CHOOSE SEALEVEL OR GROUND REFERENCE ALTITUDE READINGS. One section must be commented.
+
+  ///* SECTION 1
+  // COMMENT this section to use all sea level data, not ground reference data. Must uncomment second section. 
+  init_pressure = bmp.pressure/100;
+  init_altitude = bmp.readAltitude(SEALEVELPRESSURE_HPA);
+  x1 = read_altitude();
+  //*///END SECTION 1 
+}
+  /* SECTION 2
+  // UNCOMMENT this section to use all sea level data. Target altitude will be updated using initial altitude reading and target above ground. Must comment first section. 
+  init_pressure = SEALEVELPRESSURE_HPA;
+  init_altitude = read_altitude();
+  x1 = init_altitude;
+  target_altitude += init_altitude;
+  *///END SECTION 2 
+
+  
+
+
+  //print initial sea level altitude to file
+  myFile.print(init_altitude);
+
+  
+
+  //sets initial time, t1
+  t1 = millis();
+
+
+  
+  beep_buzz(3);//beep and buzz 
+
+
+  
+
+  //COMPUTER IS NOW ARMED
+}
+
+
+
+
+
+/////////MAIN PROGRAM///////////////MAIN PROGRAM//////////////MAIN PROGRAM///////////////MAIN PROGRAM////////////////MAIN PROGRAM////////////////MAIN PROGRAM///////////////MAIN PROGRAM/////////
+//BEGIN LOOP MAIN PROGRAM
+
+void loop() { 
+
+// ***********************STANDBY MODE************************************
+
+while(!detect_take_off()){
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+}
+
+  log_data();//logs data to sd card
+
+
+
+// ***********************ASCENDING MODE************************************
+
+while(!detect_apogee()){
+  
+  //update altitude at frequency (hz)
+  iterate_altitude();
+  
+  log_data();//logs data to sd card
+
+}
+
+  reopen_file();//saves and reopens file
+
+// ***********************FIRE 1 (DROGE CHUTE)************************************
+
+  t_droge = t2; //saves droge fire time
+  myFile.print(F("FIRE DROGE")); //logs event
+  digitalWrite(FIRE_PIN_1, HIGH); //turns on droge releay (IGN 1)
```

Comment:
I'd like to discuss what safety checks we can put around this.  For example, imagine a scenario where the execution of the program gets stuck (say, due to the SDcard malfunctioning).  The rocket could have a perfect flight with the main flight computer, but come down with one or both backup charges unfired due to a program hang.  Worse, if the program were to resume execution once on the ground, it could fire those charge(s).